File: BLH.ApproveIQ.Domain/Entities/UserProject.cs
using BLH.ApproveIQ.Domain.Primitives;

namespace BLH.ApproveIQ.Domain.Entities;

public class UserProject : AuditableEntity
{
    public Guid UserId { get; set; }
    public Guid ProjectId { get; set; }
    public string ProjectRole { get; set; } = string.Empty;

    // Navigation properties
    public User User { get; set; } = null!;
    public Project Project { get; set; } = null!;
}

File: BLH.ApproveIQ.Portal/ClientApp/src/app/data/data/b-l-h/entities/user-project.ts
import { AuditableEntity } from '../../../b-l-h/primitives/auditable-entity';
import { User } from './user';
import { Project } from './project';

export class UserProject extends AuditableEntity {
    userId: string;
    projectId: string;
    projectRole: string;
    user = new User();
    project = new Project();
}
